### PR TITLE
Fixed issue #18532: Data entry screen incorrectly assigns 0 value to …

### PR DESCRIPTION
**Pull request by adamzammit**
Fixed issue #8532: Data entry screen incorrectly assigns 0 value to empty multiflex array checkbox values

### CODE REVIEW

File: application/controllers/admin/dataentry.php [hunk 1]
```diff
@@ -1172,10 +1172,12 @@ public function editdata($subaction, $id, $surveyid)
...
                                 } else {
                                     $aDataentryoutput .= "\t<select name='{$fname['fieldname']}' class='form-control'>\n";
-                                    $aDataentryoutput .= "<option value=''>...</option>\n";
+                                    $aDataentryoutput .= "<option value=''";
+                                    if ($idrow[$fname['fieldname']] == '') {$aDataentryoutput .= " selected"; }
+                                    $aDataentryoutput .= ">...</option>\n";
                                     for ($ii = $minvalue; $ii <= $maxvalue; $ii += $stepvalue) {
                                         $aDataentryoutput .= "<option value='$ii'";
-                                        if ($idrow[$fname['fieldname']] == $ii) {$aDataentryoutput .= " selected"; }
+                                        if ($idrow[$fname['fieldname']] != '' && $idrow[$fname['fieldname']] == $ii) {$aDataentryoutput .= " selected"; }
```
Comment (Shnoulle) [inline application/controllers/admin/dataentry.php:1180]: must not be !== '' ?

else if $ii === "0" it don't work, no  ?

Comment (adamzammit) [reply application/controllers/admin/dataentry.php]: thanks I've updated it now to be more clear

File: application/controllers/admin/dataentry.php [hunk 2]
```diff
@@ -1172,10 +1172,12 @@ public function editdata($subaction, $id, $surveyid)
                                     $aDataentryoutput .= CHtml::numberField($fname['fieldname'], $idrow[$fname['fieldname']],array('step'=>'any'));
                                 } else {
                                     $aDataentryoutput .= "\t<select name='{$fname['fieldname']}' class='form-control'>\n";
-                                    $aDataentryoutput .= "<option value=''>...</option>\n";
+                                    $aDataentryoutput .= "<option value=''";
+                                    if ($idrow[$fname['fieldname']] == '') {$aDataentryoutput .= " selected"; }
```
Comment (Shnoulle) [inline application/controllers/admin/dataentry.php:1176]: Need same `$idrow[$fname['fieldname']] === ''` because `'' == '0'`

**Review by Shnoulle — CHANGES_REQUESTED** [review]
Strict compare (for choice with '0' as code)

**Review by Shnoulle — APPROVED** [review]

**Comment by adamzammit** [comment]
is this ok to merge now?

**Comment by adamzammit** [comment]
@c-schmitz these are an issue for my users are you happy for me to merge now?

**Pull request merged**